src/coreclr/nativeaot/Runtime/windows/CoffNativeCodeManager.cpp

```
@@ -164,63 +164,72 @@ static PTR_VOID GetUnwindDataBlob(TADDR moduleBase, PTR_RUNTIME_FUNCTION pRuntim
                                              PTR_RUNTIME_FUNCTION pRuntimeFunctionTable, uint32_t nRuntimeFunctionTable,
                                              PTR_PTR_VOID pClasslibFunctions, uint32_t nClasslibFunctions)
     : m_moduleBase(moduleBase),
-      m_pvManagedCodeStartRange(pvManagedCodeStartRange), m_cbManagedCodeRange(cbManagedCodeRange),
-      m_pRuntimeFunctionTable(pRuntimeFunctionTable), m_nRuntimeFunctionTable(nRuntimeFunctionTable),
-      m_pClasslibFunctions(pClasslibFunctions), m_nClasslibFunctions(nClasslibFunctions)
+    m_pvManagedCodeStartRange(pvManagedCodeStartRange), m_cbManagedCodeRange(cbManagedCodeRange),
+    m_pRuntimeFunctionTable(pRuntimeFunctionTable), m_nRuntimeFunctionTable(nRuntimeFunctionTable),
+    m_pClasslibFunctions(pClasslibFunctions), m_nClasslibFunctions(nClasslibFunctions)
 {
-}
-
-CoffNativeCodeManager::~CoffNativeCodeManager()
-{
-}
-
-static int LookupUnwindInfoForMethod(uint32_t relativePc,
-                                     PTR_RUNTIME_FUNCTION pRuntimeFunctionTable,
-                                     int low,
-                                     int high)
-{
-    // Binary search the RUNTIME_FUNCTION table
-    // Use linear search once we get down to a small number of elements
-    // to avoid Binary search overhead.
-    while (high - low > 10)
+    // max offset is beyond the range of managed methods.
+    int maxOffset = (int)((TADDR)pvManagedCodeStartRange + cbManagedCodeRange - moduleBase);
+
+    // lets build the index for the runtime table. for every granule that has elements we will have an index entry
+    uint32_t indexSize = (nRuntimeFunctionTable + FUNCTABLE_INDEX_GRANULARITY - 1) / FUNCTABLE_INDEX_GRANULARITY;
+    uint32_t indexCount = 0;
+    uint32_t* index = m_indices[indexCount++] = (uint32_t*)_aligned_malloc(indexSize * sizeof(uint32_t), INDEX_ALIGNMENT);
+    if (!index)
+        abort(); // can't allocate some modest amount of memory at startup
```

This should propagate the error instead of fail fasting.